maybe/syscall_filters.py

```diff
@@ -108,6 +109,29 @@ def substitute_open(path, flags):
         return None
 
 
+def format_mknod(path, type):
+    path = abspath(path)
+    if exists(path):
+        return None
+    elif (type & S_IFREG):
+        return "%s %s" % (T.cyan("create file"), T.underline(path))
+    elif (type & S_IFIFO):
+        return "%s %s" % (T.cyan("create named pipe"), T.underline(path))
+    else:
+        # TODO: add support for block and char special files
+        return None
+
+
+def substitute_mknod(path, type):
```

Comment:
It seems that this entire function can be reduced to

```python
return None if (format_mknod(path, type) is None) else 0
```

which has the added benefit that if support for other types is added to `format_mknod` nothing has to be changed here.